include/os/freebsd/spl/sys/ccompile.h

```diff
@@ -119,7 +119,7 @@ extern "C" {
 #define	__CONST			__sun_attr__((__const__))
 #define	__PURE			__sun_attr__((__pure__))
 
-#ifdef INVARIANTS
+#if defined(INVARIANTS) && !defined(ZFS_DEBUG)
```

Comment:
I have this change in my PR at #10634 so whichever of these goes in first, the other can be rebased.